qubesadmin/tools/qvm_start_daemon.py

```diff
@@ -114,6 +114,26 @@ def get_monitor_layout():
     return outputs
 
 
+def set_keyboard_layout(vm):
+    """Set layout configuration into features for Gui admin extension"""
+    try:
+        # Example of 'xprop -root _XKB_RULES_NAMES' output:
+        # _XKB_RULES_NAMES(STRING) = "evdev", "pc105", "fr", "oss", ""
+        xkb_rules_names = subprocess.check_output(
+            ['xprop', '-root', '_XKB_RULES_NAMES']).decode()
+        xkb_rules_names = xkb_rules_names. \
+            replace('"', '').replace('\n', '')
+        keyboard_layout = '+'.join(xkb_rules_names.split(', ')[2:])
+        vm.features['keyboard-layout'] = keyboard_layout
+
+        # Legacy
+        xkbmap = subprocess.check_output(['setxkbmap', '-print'])
+        vm.features['qubes-keyboard'] = xkbmap
```

Comment:
As discussed, setting raw `setxkbmap -print` value in features is a bad idea. This script is running in a GUI VM (which may be dom0, but doesn't need to be), and we definitely do **not** want such obscure and hard to validate value set by not-fully-trusted VM.

Reply:
To be sure, one problem is the format of the value and because it goes to __features__? Which are __local__ values instead of in-VM values (qubesdb) right ?

Reply:
So one step behind again in trying to reconstruct this value? Also, all of those troubles come from the attempt of keeping legacy settings in R4.0 templates. Why simply not trying to make deprecated `/qubes-keyboard` and use in template to use `/keyboard-layout` for R4.0?